homeassistant/components/onkyo/media_player.py

```diff
@@ -115,395 +196,350 @@
-            current_source_tuples = (current_source_raw[0], (current_source_raw[1],))
-        else:
-            current_source_tuples = current_source_raw
+    @callback
+    def _parse_source(self, source_raw):
+        values = _parse_onkyo_tuple(source_raw)
 
-        for source in current_source_tuples[1]:
+        for source in values:
             if source in self._source_mapping:
-                self._current_source = self._source_mapping[source]
+                self._source = self._source_mapping[source]
                 break
-            self._current_source = "_".join(current_source_tuples[1])
-        self._muted = bool(mute_raw[1] == "on")
-        if preset_raw and self._current_source.lower() == "radio":
-            self._attributes[ATTR_PRESET] = preset_raw[1]
-        elif ATTR_PRESET in self._attributes:
-            del self._attributes[ATTR_PRESET]
-        if self._supports_volume:
-            # AMP_VOL/MAX_RECEIVER_VOL*(MAX_VOL/100)
-            self._volume = (
-                volume_raw[1] / self._receiver_max_volume * (self._max_volume / 100)
-            )
-
-    @property
-    def supported_features(self):
-        """Return media player features that are supported."""
-        if self._supports_volume:
-            return SUPPORT_ONKYO
-        return SUPPORT_ONKYO_WO_VOLUME
-
-    def turn_off(self):
-        """Turn the media player off."""
-        self.command(f"zone{self._zone}.power=standby")
-
-    def set_volume_level(self, volume):
-        """
-        Set volume level, input is range 0..1.
-
-        However full volume on the amp is usually far too loud so allow the user to specify the upper range
-        with CONF_MAX_VOLUME.  we change as per max_volume set by user. This means that if max volume is 80 then full
-        volume in HA will give 80% volume on the receiver. Then we convert
-        that to the correct scale for the receiver.
-        """
-        # HA_VOL * (MAX VOL / 100) * MAX_RECEIVER_VOL
-        self.command(
-            f"zone{self._zone}.volume={int(volume * (self._max_volume / 100) * self._receiver_max_volume)}"
-        )
+            self._source = "_".join(values)
+
+    @callback
+    def _parse_sound_mode(self, sound_mode_raw):
+        values = _parse_onkyo_tuple(sound_mode_raw)
+
+        # If the selected sound mode is not available, N/A is returned
+        # so only update the sound mode when it is not N/A
+        if "N/A" not in values:
+            for sound_mode in values:
+                if sound_mode in SOUND_MODE_REVERSE_MAPPING:
+                    self._sound_mode = SOUND_MODE_REVERSE_MAPPING[sound_mode]
+                    break
+                self._sound_mode = "_".join(values)
+
+    @callback
+    def _parse_audio_inforamtion(self, audio_information_raw):
+        values = _parse_onkyo_tuple(audio_information_raw)
+        if "N/A" not in values:
+            info = {
+                "format": _tuple_get(values, 1),
```

Comment:
Ideally the pypi package is handling presenting this data to Home Assistant in a more consumable form that avoid using the magic numbers below.